src/utils/helpers/revalidateAll.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  ApplicationElementStates,
+  ResponsesByCode,
+  RevalidateResult,
+  ValidityFailure,
+} from '../types'
+import { defaultValidate } from '../../formElementPlugins/defaultValidate'
+
+export const revalidateAll = async (
+  elementsState: ApplicationElementStates | undefined,
+  responsesByCode: ResponsesByCode | undefined,
+  strict = true,
+  shouldUpdateDatabase = true
+): Promise<RevalidateResult> => {
+  const validate = defaultValidate // To-Do: import custom validation methods
+
+  // console.log('Elements', elementsState)
+  // console.log('responses', responsesByCode)
+  if (elementsState) {
+    const elementCodes = Object.keys(elementsState).filter(
+      (key) =>
+        responsesByCode && // Typescript requires this
+        elementsState[key].category === 'QUESTION' &&
+        elementsState[key].isVisible === true &&
+        // Strict/Loose validation logic:
+        ((strict && elementsState[key].isRequired) ||
+          (strict && responsesByCode[key]?.isValid !== null) ||
+          (!strict && responsesByCode[key]?.isValid !== null))
```

Comment:
```suggestion
          (responsesByCode[key]?.isValid !== null))
```

Reply:
This logic was doing my head in. Just to be clear, it's supposed to be KEEPING for validation:
- if strict: all required and all not null
- if not strict: all not null.

That's right, isn't it?